python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py

```diff
@@ -0,0 +1,388 @@
+"""Torch Module for Relational graph convolution layer using the
+aggregation primitives in cugraph-ops"""
+# pylint: disable=no-member, arguments-differ, invalid-name, too-many-arguments
+import math
+
+import torch as th
+from torch import nn
+
+try:
+    from pylibcugraphops.legacy.aggregators.node_level import (
+        agg_hg_basis_post_bwd_int32,
+        agg_hg_basis_post_bwd_int64,
+        agg_hg_basis_post_fwd_int32,
+        agg_hg_basis_post_fwd_int64,
+    )
+    from pylibcugraphops.legacy.structure.graph_types import (
+        message_flow_graph_hg_csr_int32,
+        message_flow_graph_hg_csr_int64,
+    )
+except ModuleNotFoundError:
+    raise ModuleNotFoundError(
+        "dgl.nn.CuGraphRelGraphConv requires pylibcugraphops to be installed."
+    )
+
+CUDA_SM_PER_BLOCK = 49152
```

Comment:
Maybe add a comment and link to a documentation about this magic number? If possible.

Reply:
I have left some explanation there. We are working on an improvement in the next release of cugraph-ops so that this memory allocation size check actually happens in cugraph-ops.